src/lib.rs

```diff
@@ -49,22 +49,26 @@ impl<Pat> Grammar<Pat> {
             fn fold_repeat_many(
                 &mut self,
                 elem: RuleWithNamedFields<Pat>,
-                sep: Option<RuleWithNamedFields<Pat>>,
+                sep: Option<(RuleWithNamedFields<Pat>, SepKind)>,
             ) -> RuleWithNamedFields<Pat> {
-                elem.fold(self).repeat_many(Some(
-                    sep.map_or_else(empty, |sep| self.whitespace.clone() + sep.fold(self))
-                        + self.whitespace.clone(),
-                ))
+                elem.fold(self).repeat_many(Some((
+                    sep.map_or_else(empty, |(sep, _)| {
+                        self.whitespace.clone() + sep.fold(self)
+                    }) + self.whitespace.clone(),
+                    SepKind::Simple
```

Comment:
This loses the kind.

Reply:
Darn my rebasing mess making history hard to find

I had it [this way previously](https://github.com/rust-lang/gll/blob/4ac734b5d242baed51490eae54fdd28c2f89f435/src/grammar.rs#L76-L85). Looking back at it, that probably is wrong, as it just makes the trailing-ness disappear when using whitespace if I'm reading it correctly 😅